crates/sui-core/src/unit_tests/authority_tests.rs

```diff
@@ -1535,6 +1536,239 @@ async fn test_handle_sponsored_transaction() {
     );
 }
 
+#[tokio::test]
+async fn test_move_sponsored_transaction() {
```

Comment:
We have --sponsor in the transactional test adapter. I feel like we probably should have written the test there instead of here

Reply:
done